LabExT/Tests/Wafer/Export_data_test.py

```diff
@@ -0,0 +1,126 @@
+"""
+
+import json
+from os.path import dirname, join, exists
+from os import remove
+from unittest.mock import Mock
+import pytest
+import uuid
+import time
+
+from LabExT.Tests.Utils import TKinterTestCase
+from LabExT.Utils import setup_user_settings_directory
+from LabExT.Exporter.Formats.ExportCSV import ExportCSV
+from LabExT.Exporter.Formats.ExportHDF5 import ExportHDF5
+from LabExT.Exporter.Formats.ExportJSON import ExportJSON
+from LabExT.Exporter.ExportWizard import ExportWizard
+from LabExT.ViewModel.Utilities.ObservableList import ObservableList
+
+        
+class MockTKVar:
+    def __init__(self, value):
+        self.value = value
+
+    def set(self, value):
+        self.value = value
+
+    def get(self):
+        return self.value
+        
+    __repr__ = __str__ = lambda self: self.value
+            
+class ExportDataTest(TKinterTestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        setup_user_settings_directory(makedir_if_needed=True)
+
+        self.expm = Mock()
+        self.expm.export_format_api.export_formats = {
+            "ExportJSON": ExportJSON,
+            "ExportCSV": ExportCSV,
+            "ExportHDF5": ExportHDF5,
+        }
+
+        file_name = join(dirname(dirname(__file__)), "example_measurement.json")
+        self.expm.exp.measurements = ObservableList()
+
+        with open(file_name) as f:
+            meas_dict = json.load(f)
+
+            # mock the loading of the measurement
+            # this is normally done in the ExperimentManager
+            meas_dict["file_path_known"] = file_name
+
+            for k in ["timestamp start", "timestamp", "timestamp end"]:
+                if k in meas_dict:
+                    meas_dict["timestamp_known"] = meas_dict[k]
+                    break
+          
+            for k in ["measurement name", "name"]:
+                if k in meas_dict:
+                    meas_dict["name_known"] = meas_dict[k]
+                    break
+           
+            meas_dict["measurement id long"] = uuid.uuid4().hex
+
+            for k in ["timestamp iso start", "timestamp_known"]:
+                if k in meas_dict:
+                    meas_dict["timestamp_iso_known"] = meas_dict[k]
+                    break
+
+            self.expm.exp.measurements.append(meas_dict)
+
+    def setup_window(self):
+        return ExportWizard(master=self.root, experiment_manager=self.expm)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_create_window_instantiate_steps(self):
+        window = self.setup_window()
+        self.assertIsInstance(window.source_config_steps_insts["JavaScript Object Notation (.json)"], ExportJSON)
+        self.assertIsInstance(window.source_config_steps_insts["Comma-Separated Values (.csv)"], ExportCSV)
+        self.assertIsInstance(window.source_config_steps_insts["Hierarchical Data Format (.hdf5)"], ExportHDF5)
+
+    @pytest.mark.flaky(reruns=0)
+    def test_csv_format(self):
```

Comment:
Good job in finding out how to test-click through the GUI. I'd recommend adding some more asserts after every click to actually do testing.